Lidi branch abstract

lmanov1 opened the pull request.

GuyYar1 committed:
Support_Intel_NvidiaCuda_based.

GuyYar1 committed:
Update

lmanov1 committed:
Initial environment and setup, documented in README.md

Setup: Fixes for GPU environment (check_hardware_and_install.py ) - Automatic GPU detection and installation of required packages (CUDA)
poetry.lock and pyproject.toml are updated with the new dependencies
Dataloader.py - Fixes
main.py - Should be separated from check hardware and environment installation
README.md - Updated with the new setup and environment installation instructions
.gitignore - Updated with the new files and directories to ignore (including data/raw locations , idea folder, kaggle.json)
TBD - secret management on github

lmanov1 committed:
Refactor some of the code for improved readability and maintainability

lmanov1 committed:
initial

lmanov1 committed:
working dataloaders and basic models + cflassification; introducing the abstract class for eye desieses classification

lmanov1 committed:
added project blueprint and classes description - first draft , to be extended

lmanov1 committed:
same parameters after train

GuyYar1 commented:
Hi Lidi

i pulled my branch and few comments:

1. i have conflict that might suggest to upgrade to numpy 2.0.2
    can you upgrade and push changes to my branch.
2. i noticed that you type this comment in the
"code/Util/check_hardware_and_install.py"  :  +    # tensorflow-cpu and
tensorflow-gpu are not supported started from 2.17.0. Instead , use
tensorflow
+    # and in addition install CUDA if relevant so that tensorflow will be
able to leverage it"


Question: what i should do here - can i talk with you about it ?

3.  i think that the correct way to work here: each will commit to their
branch and sync with main branch when necessary.

when you commit to my branch it causes to conflicts that i must solve
before i finish my work.

4. I noticed that you created a new branch as we spoke about abstract, i
will review it and update you.

Many thanks
Guy


On Sat, Oct 12, 2024, 20:28 Lidia Manov ***@***.***> wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2
> Commit Summary
>
>    - d118537
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/d1185374c1f06e64bb493f5872f9f7324ac06f28>
>    Support_Intel_NvidiaCuda_based.
>    - 0b5eff5
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/0b5eff5ccc4d3fb062eca78cdd0382a3d6d3ade2>
>    Update
>    - 9734f18
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/9734f18e273db0a7c7a2a17948ff8564ecdf47bd>
>    Initial environment and setup, documented in README.md
>    - dc8ad28
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/dc8ad28d875d66e16814c550b1092693b11a9d05>
>    Refactor some of the code for improved readability and maintainability
>    - e28aebf
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/e28aebf3b711f911be74a4ebdb1137ee133dec7e>
>    initial
>    - 0228c6f
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/0228c6fd413437886fef21c0a7c3bc263ffd9d75>
>    working dataloaders and basic models + cflassification; introducing the
>    abstract class for eye desieses classification
>    - b27a3dd
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/b27a3dd3215bbef86ec500fe7bb1c18dc2049b46>
>    added project blueprint and classes description - first draft , to be
>    extended
>    - 7e4922f
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/commits/7e4922f9ea65a9f470af1edfd0e4c1dabe8db78d>
>    same parameters after train
>
> File Changes
>
> (20 files
> <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files>
> )
>
>    - *M* .gitignore
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (4)
>    - *A* .idea/aws.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-b1c99adeb270feceb7dc928e5b7c94b8ceb69c592b8402e942755f56a89f4cc8>
>    (11)
>    - *A* .idea/inspectionProfiles/Project_Default.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-bbf0d02ebcf6764cae92fa4045a2d6f15f8a479cd1aa2ab4d1fd435a3aa74bbd>
>    (16)
>    - *A* .idea/inspectionProfiles/profiles_settings.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-9fb34448452829ed3cd715b49c0f4494fe0b97a4aef2a5b23e12c486ed932273>
>    (6)
>    - *A* .idea/misc.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-7ea1cd7bd5d16299a0abd33128e8d0a8bc04146f6b00c07bcd3cc0aad813c229>
>    (7)
>    - *A* .idea/vcs.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-a6dab70bd0e9ffc91d8419ad52225baff3a3e7c9d8e1724f7ed6d96b661d804a>
>    (6)
>    - *A* .idea/workspace.xml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-9f043e4e89d784b579b6d75ea8704b232f9475fa5b3afd11f805fdf8e74c675c>
>    (81)
>    - *M* README.md
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (156)
>    - *M* code/README.md
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-42c2e34932b0586c4a07e17402fddcd832a08c61384db8cfcce66e9b16211295>
>    (1)
>    - *A* code/Util/MultiPlatform.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-9fae35df0a61b9a74fb5790251e13e6a3810431c76ff9f3aac174ecbce4c1a95>
>    (27)
>    - *A* code/Util/check_hardware_and_install.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-e0ac3cf9232ac5a3551561045897e1769484bdb20733372562d057471704c44b>
>    (76)
>    - *A* code/Util/kaggle.json
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-825a375dfcd407012ff53e6ff2352c43e43808c5244dd6313afcdc106a3c9811>
>    (0)
>    - *A* code/data/Dataloader.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-0197517a95305b2179a78682c387449982ac1db5d36a00b300031c35961fe7df>
>    (80)
>    - *M* code/data/data_preparation.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-a3d65fd0d69b79db85f418bcae81b0cb2d8975bb53052cd65d8782fdb5ee9c7d>
>    (169)
>    - *A* code/main.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-7064e9412d3433452f4affd6fe285500ff3a2a4328232d57e3e86cf437b086b4>
>    (125)
>    - *M* code/models/inference_model.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-bde8b4ef2ba7d18e84022038389f480865c0f1c563b2508c7ae9a1c2263d2f66>
>    (23)
>    - *M* code/models/train_model.py
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-d1abae161715a1a49ead4864a719320a5d90f97c87ed247b82483e7a404e6ebc>
>    (80)
>    - *M* poetry.lock
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-f53a023eedfa3fbf2925ec7dc76eecdc954ea94b7e47065393dbad519613dc89>
>    (98)
>    - *M* pyproject.toml
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711>
>    (12)
>    - *M* requirements.txt
>    <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
>    (117)
>
> Patch Links:
>
>    -
>    https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2.patch
>    -
>    https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lmanov1/DL_DiabeticRetinopathyStagePrediction/pull/2>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7TJ5L3V3DQ6V4GGJJT3VJ3Z3FL5RAVCNFSM6AAAAABP2S5VD2VHI2DSMVQWIX3LMV43ASLTON2WKOZSGU4DGMRXGQ2DENI>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

lmanov1 commented:
1&2 - the final setup as described in README.md - 'Getting started' section.
I tested this on another (Linux) comp without GPU—from scratch—and the installation worked seamlessly. Still, there is some probability it may have issues on Windows. I would appreciate it if you could test it at your premises.
3  - yes, you're right - I branched out from your branch because I wanted to have all your new code + my fixes on it  as a base to start with (as it is still not merged to the main). Hopefully, only once.

lmanov1 committed:
Models training flow tuning - not running pretrained model again if its was previously saved (heavy run)

Saving models in torch format in data/output
Loading pretrained weigths into the train model (transfer learning)
Simple statiststics that can be printed to a console (crash fixed)
not commiting saved model files (.gitignore)
handling different linux distros and package manager types to install cuda on gpu-enabled systems
added util function to print time in human readable format (tbd - python exitisting functions?)

lmanov1 committed:
learner and statistics on model - fix

lmanov1 committed:
Classifiers with multiple inheritance - to allow code reuse

(training/evaliation) across different models that inherit same basic
methods
README.md updated
main flow updated

lmanov1 committed:
poetry init missed in getting started

lmanov1 committed:
readme again

lmanov1 commented on my_code/main.py:
```diff
@@ -0,0 +1,414 @@
+
+
+# Check for NVIDIA GPU and install the correct TensorFlow version
+def is_nvidia_gpu_available():
```
@GuyYar1  - can you please explain why install tensor flow (or anything else) from here?  In general , installer and the functionality should not go together. package installation is a once-done task and should not be done each time we run a model. This is what I think. Additionally, as I explained earlier,  tensorflow-gpu  (last - tensorflow-gpu 2.12.0 not 2.17) package is deprecated and does not exists since 2022 , instead , we need to install just tensorflow which works in both modes (CPU, gpu) : https://pypi.org/project/tensorflow-gpu/
All this code was in the Util section (fixed) , could you explain please why it is moved here?

GuyYar1 replied:
We decided that it will run on one platform using wsl no need to multiple platform support - so N\A

GuyYar1 reviewed (approved):
Ok.

lmanov1 committed:
removed malformed file - this one caused GUY's problems possibly

lmanov1 committed:
GUY's use os.path.sep instead of code

lmanov1 committed:
this addition should enable importing code from these directories style from models.train_model import *

lmanov1 committed:
compatibility to windows versions

lmanov1 committed:
clarifications for how to run on wndows
